src/foam/nanos/http/LiveScriptBundler.java

```diff
@@ -0,0 +1,396 @@
+      final int STATE_SKIP_WS = 5;
+      int state = STATE_FIND_NAME;
+      int stateStack[] = {0, 0, 0}; // nested control flow makes it fun
+      int stateStackPtr = 0;
+
+      for ( boolean done = false; !done; ) {
+        switch (state) {
+          case STATE_SKIP_WS:
+            if (
+              line.charAt(pos) == ' '  ||
+              line.charAt(pos) == '\t' ||
+              line.charAt(pos) == '\r' // unlikely, but just to be safe
+              // '\n' is not possible here
+            ) {
+              pos++;
+            } else {
+              // Go to the next state which the previous state
+              // wanted us to go to.
+              state = stateStack[--stateStackPtr];
+            }
+            break;
+          case STATE_FIND_NAME:
+            if ( line.substring(pos).startsWith("name:") ) {
+              pos += 5;
+              state = STATE_EAT_NAME;
+            } else if ( line.substring(pos).startsWith("\"name\":") ) {
+              pos += 7;
+              state = STATE_EAT_NAME;
+            } else {
+              pos++;
+            }
+            break;
+          case STATE_EAT_NAME:
+            // Remember where we were after skipping whitespace
+            stateStack[stateStackPtr++] = STATE_EAT_NAME_2;
+            state = STATE_SKIP_WS;
+            break;
+          case STATE_EAT_NAME_2:
+            char term = line.charAt(pos);
+            pos++;
+            int nameStart = pos;
+            boolean inEscape = false;
+            for ( ;;pos++ ) {
+              char thisChar = line.charAt(pos);
+              if ( inEscape ) {
+                inEscape = false;
+              } else {
+                if ( thisChar == '\\' ) {
+                  inEscape = true;
+                } else if (thisChar == term) {
+                  break;
+                }
+              }
+            }
+            name = line.substring(nameStart, pos);
+            pos++;
+            // Remember where we were after skipping a comma
+            stateStack[stateStackPtr++] = STATE_FIND_FLAGS_OR_TERM;
+            state = STATE_SKIP_COMMA;
+            break;
+          case STATE_SKIP_COMMA:
+            // Remember where we were after skipping whitespace
+            stateStack[stateStackPtr++] = STATE_SKIP_COMMA_2;
+            state = STATE_SKIP_WS;
+            break;
+          case STATE_SKIP_COMMA_2:
+            if ( line.charAt(pos) == ',' ) {
+              pos++;
+            }
+            state = stateStack[--stateStackPtr];
+            break;
+          case STATE_FIND_FLAGS_OR_TERM:
+            // For now, don't actually "find" the flags;
+            // instead, take the whole line after the file name
+            // to later see if it contains "web" anywhere.
+            flags = line.substring(pos);
+            done = true;
+        }
+      }
+
+      name += ".js";
+      paths.add(name);
+    }
+
+    return paths;
+  }
+
+  private void log_(String evt, String msg) {
```

Comment:
Why do this and not just use nanos logger?

Reply:
This was updated before LiveScriptBundler was merged. Not sure what happened to this branch but it's a disaster now so I created a new one.